service/create/service.go

```diff
@@ -186,18 +220,144 @@ func (s *Service) Boot() {
 	})
 }
 
-func (s *Service) runMachines(machines []node.Node, clusterName, prefix string) error {
+func (s *Service) runMachines(spec awstpr.Spec, clusterName string, prefix string) error {
+	var (
+		machines        []node.Node
+		awsMachines     []tpraws.Node
+		cloudconfigPath string
+	)
+
+	switch prefix {
+	case prefixMaster:
+		machines = spec.Masters
+		awsMachines = spec.Aws.Masters
+		cloudconfigPath = s.cloudconfigMasterPath
+	case prefixWorker:
+		machines = spec.Workers
+		awsMachines = spec.Aws.Workers
+		cloudconfigPath = s.cloudconfigWorkerPath
+	default:
+		return microerror.MaskAny(fmt.Errorf("invalid prefix %q", prefix))
+	}
+
+	if len(machines) != len(awsMachines) {
+		return microerror.MaskAny(fmt.Errorf("mismatched number of %q machines in the 'spec' and 'aws' sections: %d != %d",
+			prefix,
+			len(machines),
+			len(awsMachines)))
+	}
+
 	for no, machine := range machines {
 		name := fmt.Sprintf("%s-%d", prefix, no)
-		if err := s.runMachine(machine, clusterName, name); err != nil {
+		m := awsNode{
+			Node:    machine,
+			AwsInfo: awsMachines[no],
+		}
+		if err := s.runMachine(m, spec, clusterName, cloudconfigPath, name); err != nil {
 			return microerror.MaskAny(err)
 		}
 	}
 	return nil
 }
 
-func (s *Service) runMachine(machine node.Node, clusterName, name string) error {
-	instances, err := s.ec2Client.DescribeInstances(&ec2.DescribeInstancesInput{
+const (
+	roleName                 = "EC2-K8S-Role"
+	policyName               = "EC2-K8S-Policy"
+	profileName              = "EC2-DecryptTLSCerts"
+	assumeRolePolicyDocument = `{
+		"Version": "2012-10-17",
+		"Statement": {
+			"Effect": "Allow",
+			"Principal": {
+				"Service": "ec2.amazonaws.com"
+			},
+			"Action": "sts:AssumeRole"
+		}
+	}`
+	policyDocumentTempl = `{
+		"Version": "2012-10-17",
+		"Statement": [
+			{
+				"Effect": "Allow",
+				"Action": "kms:Decrypt",
+				"Resource": %q
+			}
+		]
+	}`
```

Comment:
I think it would be good to have these as public variables to have it in the godoc.